terraform/api.tf

```diff
@@ -65,61 +65,3 @@ data "aws_iam_policy_document" "api_dandisets_bucket" {
     ]
   }
 }
-
-resource "aws_iam_role" "write_public_dataset" {
-  name               = "write-public-dataset"
-  description        = "Allows EC2 instances to call AWS services on your behalf."
-  assume_role_policy = data.aws_iam_policy_document.write_public_dataset.json
-  inline_policy {
-    name   = "write-public-dataset"
-    policy = data.aws_iam_policy_document.write_public_dataset_inline.json
-  }
-  managed_policy_arns = [
-    "arn:aws:iam::aws:policy/AmazonS3FullAccess",
-  ]
-}
-
-data "aws_iam_policy_document" "write_public_dataset" {
-  version = "2012-10-17"
-  statement {
-    actions = ["sts:AssumeRole"]
-    principals {
-      type        = "Service"
-      identifiers = ["ec2.amazonaws.com"]
-    }
-  }
-}
-
-data "aws_iam_policy_document" "write_public_dataset_inline" {
-  version = "2012-10-17"
-  statement {
-    actions   = ["sts:AssumeRole"]
-    effect    = "Allow"
-    resources = [aws_iam_role.sponsored_dandi_writer.arn]
-  }
-}
-
-
-resource "aws_iam_role" "dandi_girder" {
-  name               = "dandi-girder"
-  assume_role_policy = data.aws_iam_policy_document.dandi_girder.json
-  inline_policy {
-    name   = "write-public-dataset"
-    policy = data.aws_iam_policy_document.write_public_dataset_inline.json
-  }
-  managed_policy_arns = [
-    "arn:aws:iam::aws:policy/AmazonEC2FullAccess",
-    "arn:aws:iam::aws:policy/AmazonS3FullAccess",
-  ]
-}
-
-data "aws_iam_policy_document" "dandi_girder" {
-  version = "2012-10-17"
-  statement {
-    actions = ["sts:AssumeRole"]
-    principals {
-      type        = "Service"
-      identifiers = ["ec2.amazonaws.com"]
-    }
-  }
-}
```

Comment:
I have a hypothesis that this role is actually used somewhere in `girder.tf` and we were mistaken to add it here in the first place. I agree it shouldn't be here, but I think just merging this will delete it, which might mess some stuff up.

Reply:
I think this was previously used to permit access for an additional Girder 3 assetstore. If you can verify that there is not an actively-used assetstore on the Girder 3 instance that uses this (possibly via an instance profile), it should be safe to delete.

Reply:
I'm not totally following, but there is an instance profile that is used by Girder 3 to write to the sponsored bucket.

I believe the EC2 Girder 3 runs on was launched with an instance profile, and we also pushed a [config file](https://github.com/dandi/dandiarchive/blob/master/ansible/files/config) onto that instance so the instance could talk to the sponsored bucket. I included a link to the config file because that might help you hunt down any further needed info.

Reply:
As soon as Girder 3 no longer needs to *write* to the sponsored bucket, there should be no problems; reading is always allowed, since this is public.

There's no urgency to merge this PR, but it would be good for maintainability to get rid of all these excess resources at some point.